### DISCUSSION TITLE
March 15th 2024 - Landmark Release 🥳

### DISCUSSION THREAD

**Discussion by rphlo** [karttapullautin/karttapullautin#60] [Announcements]
`knolldetector` function is now converted to rust.

This mean all the functions to convert a .LAZ, .LAS, or .XYZ with default parameters are now implemented in rust.

You can now run use `rusty-pullauta tile.laz` instead of `perl pullauta tile.laz`  
For batch mode you can use `rusty-pullauta` instead of `perl pullauta`

Notes:
 - Rusty-Pullauta still doesn't include the functions to process zip files, so you'll still need the perl script if you intend to use those.
 - Rusty-pullauta also does not support the option `vegemode=1`

<hr /><em>This discussion was created from the release <a href='https://github.com/rphlo/rusty-pullauta/releases/tag/20240315a'>March 15th 2024 - Landmark Release 🥳</a>.</em>

**Comment by stefankinell** [comment]
Very cool @rphlo !!! Thanks for keep working on the tool. It is much appreciated. And fun to be part of testing along the way.

**Reply by rphlo** [reply]
@stefankinell and other rusty pullauta users, do you use the shapefile feature of pullauta?, or is it as I understand Finland specific piece of code?
It is 2500 lines of code, so if it is finland specific I could see myself not translating it to rust.
Also is there other commands offered by pullauta (e.g. "pngmerge" or "dxfmerge") not yet translated to rust that you use?

**Reply by stefankinell** [reply]
I have never used the zip-file feature. But I tend to use kartapullautin only to generate vector files and background vegetation images, not using the ready map output files as much. I think the zip-files make sense only if you are interested in getting a ready map-png. 

For the other commands - I do use pngmerge and more often pngmergevege. But I have no problem using the perl-version for these commands. I used to run makevegenew but now with the new vegetation only in batch mode I stoped using that. Can not come up with other things I normally use.

**Reply by rphlo** [reply]
`makevegenew` was transcribed to rust already.
`pngmerge` and `pngmergevege` are not, could be next thing to do...

**Reply by NicoRio42** [reply]
@rphlo the shapefile feature is not Finland specific, you can use any shapefile as long as you provide a "mapping file" in the "vectorconf" field of the karttapullautin configuration file. I used this functionality to display vector features from OpenStreetMap and IGN data on mapant.fr.
Awesome work by the way!

**Reply by rphlo** [reply]
@NicoRio42 true, I noticed that, I will first fix the batch mode then start working on the shape file processing. I'll probably not write the finland specific code but only implement the vectorconf mode.

**Comment by rphlo** [comment]
Since v2.0.0 all code is running with rust.
